internal/metrics/metrics.go

```diff
@@ -24,14 +24,17 @@ const (
 )
 
 // Metrics represents a monitoring API such as Stackdriver.
-//
-// Latency returns the request latency after applying the specified series
-// aligner and cross series reducer. The result is in milliseconds.
-//
-// Error rate gets all the server responses. It calculates the error rate by
-// performing the operation (5xx responses / all responses).
 type Metrics interface {
```

Comment:
From the standpoint of making the metrics in use flexible, I'm surprised to see `Metrics` as an interface.

I expected to see:

```
var metrics []Metric
type Metric interface
type RequestCount Metric
```

Will leave this as a future consideration, on second glance it looks like this was a previous decision :)

Reply:
Yeah imo this should be named MetricsProvider, as it is meant to be an interface for Stackdriver and Prometheus like apis

Reply:
Oh, I think the naming is what make it confusing. The interface is meant to facilitate the creation of different types of metrics providers (Stackdriver, Prometheus, etc). Might be worth renaming this interface. What do you think?

Reply:
Seems like @ahmetb commented at the same time as me. I've created an issue about this #25 . I will rename this interface in a different PR